File: 0735-asteroid-collision/0735-asteroid-collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& asteroids) {
        vector<int> result;
        int length = asteroids.size();
        for (int i = 0; i < length; i++) {
            bool blows = false;
            while (result.size() && asteroids[i] < 0 && result.back() > 0 && !blows) {
                if (abs(asteroids[i]) < result.back()) {
                    blows = true;
                } else {
                    if (abs(asteroids[i]) == result.back()) {
                        blows = true;
                        result.pop_back();
                    } else {
                        result.pop_back();
                    }
                }
            }
            if (!blows) {
                result.push_back(asteroids[i]);
            }
        }
        return result;
    }
};

File: 0735-asteroid-collision/0735-asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        res = list()
        for ast in asteroids:
            blows = False
            while res and ast < 0 and res[-1] > 0 and not blows:
                if abs(ast) < res[-1]:
                    blows = True
                else:
                    if abs(ast) == res[-1]:
                        blows = True
                        res.pop()
                    else:
                        res.pop()
            if not blows:
                res.append(ast)
        return res